Sources/Helpers/CellItemFactory.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  CellItemFactory.swift
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+public protocol CellItemFactory {
+    
+    /// Returns an array of cell items
+    ///
+    /// - Parameters:
+    ///    - array: an array of objects to create cell items for them
+    func makeCellItems(array: [Any]) -> [CollectionViewCellItem]
```

Comment:
Maybe protocol with associated type will better? I don't like `Any` type really

Reply:
No, it is common abstraction for all future cellItem factories. for example, `ComplexCellItemFactory` provides an ability to combine a number of associated types, that can be not nested from common generic type